File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    int sumOfLeftLeaves(TreeNode* root) {
        stack<pair<TreeNode*, bool>> st;
        st.push({root,false});
        int ans =0;
        while(!st.empty()){
            auto [cur, isLeft] = st.top(); st.pop();
            if(!cur->left && !cur->right && isLeft){
                ans+= cur->val;
            }
            if(cur->left){
                st.push({cur->left,true});
            }
            if(cur->right){
                st.push({cur->right,false});
            }
        }
        return ans;
    }
};

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    private int isLeftTrue(TreeNode root, boolean isLeft){
        if(root == null) return 0;
        if(root.left == null && root.right == null && isLeft == true){
            return root.val;
        }
        return isLeftTrue(root.left, true) + isLeftTrue(root.right, false);
    }
    public int sumOfLeftLeaves(TreeNode root) {
        return isLeftTrue(root, false);
    }
}